src/olympia/devhub/tests/test_views.py

```diff
@@ -1331,21 +1331,14 @@ def test_no_servererror_on_missing_version(self):
     def test_not_a_valid_xpi(self, run_addons_linter_mock):
         run_addons_linter_mock.return_value = json.dumps(self.validation_ok())
         self.upload_file('unopenable.xpi')
-        # We never even reach the linter (we can't: because we're repacking
-        # zip files, we should raise an error if the zip is invalid before
-        # calling the linter, even though the linter has a perfectly good error
-        # message for this kind of situation).
         assert not run_addons_linter_mock.called
         upload = FileUpload.objects.get()
         response = self.client.get(
             reverse('devhub.upload_detail', args=[upload.uuid.hex, 'json']))
         data = json.loads(force_text(response.content))
         message = [(m['message'], m.get('fatal', False))
                    for m in data['validation']['messages']]
-        # We do raise a specific error message explaining that the archive is
-        # not valid instead of a generic exception.
```

Comment:
@diox I reverted changes you made here when you combined the repack and linter tasks

Reply:
mmm you shouldn't need to change this test if everything still works. We *do* want the specific message instead of the generic one. The comment above should also still apply: if the xpi is not a valid zip the chain should abort and the linter should never be called.

Reply:
mmh, interesting. I wrapped the repack task with `validation_task` (otherwise, everything gets broken...) but it is likely why we do not get the specific error. It likely swallows the exception, no?

Reply:
But `validation_task` *is* the one that returns that `Invalid or corrupt add-on file` message, if `BadZipfile` is raised by the task it's wrapping. And this should be the right exception to catch, since it was working before ? Something is not right.

Reply:
Yeah, I don't get it either to be honest.. I'll investigate

Reply:
It might be precisely the issue we wanted to avoid by upgrading Celery - it might also be only in tests, since we're running them in eager mode and that messes everything up... do try on your actual env

Reply:
I think I get it:

I added `@validation_task` to the `repack` task, so that we can use its error handling and return the `Invalid or corrupt add-on file` message. The problem is that it means the exception is handled and we do not call `on_error()` because there is not need (the exception has been handled). Instead, `repack` returns the `results` but because `validate_upload` is declared as immutable, the `results` from `repack` are lost.

That is why `validate_upload` is fired.

If we do not use `@validation_task` on the `repack` task, we cannot return a specific message. I forgot about it, but that's what I assumed last night... and that's why I changed this test case.

If we want to return a specific message, I suppose we need to move the logic from `@validation_task` to the `Validator.error_handler` method, assuming raising in a chain calls this handler (which was not the case in Celery < 4.2 if I understood correctly).

Reply:
Yeah, I think that makes sense. We do want the specific error message, so now that we have a chain the logic for that must be in the error handler.